### PR TITLE
add async to repo and commands; add ImageSupport

### PR DESCRIPTION
**Pull request by ndoriomedov**

### CODE REVIEW

**Review by BelehovEgor — APPROVED** [review]

File: src/ImageService.Broker/Consumers/CreateImagesConsumer.cs [hunk 1]
```diff
@@ -79,37 +79,42 @@ private object CreateUserImages(ICreateImagesRequest request)
...
       DbImageUser dbImageUser;
       DbImageUser dbPrewiewImageUser;
-      string resizedContent;
+      (bool isSuccess, string resizedContent, string extension) resizeResult;
 
       foreach (CreateImageData createImage in request.Images)
       {
         dbImageUser = _dbImageUserMapper.Map(createImage);
-        resizedContent = _resizeHelper.Resize(createImage.Content, createImage.Extension);
+        resizeResult = await _resizeHelper.Resize(createImage.Content, createImage.Extension);
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/CreateImagesConsumer.cs:87]: Resize async?

File: src/ImageService.Broker/Consumers/CreateImagesConsumer.cs [hunk 2]
```diff
@@ -120,37 +125,42 @@ private object CreateProjectImages(ICreateImagesRequest request)
...
       DbImageProject dbImageProject;
       DbImageProject dbPrewiewImageProject;
-      string resizedContent;
+      (bool isSuccess, string resizedContent, string extension) resizeResult;
 
       foreach (CreateImageData createImage in request.Images)
       {
         dbImageProject = _dbImageProjectMapper.Map(createImage);
-        resizedContent = _resizeHelper.Resize(createImage.Content, createImage.Extension);
+        resizeResult = await _resizeHelper.Resize(createImage.Content, createImage.Extension);
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/CreateImagesConsumer.cs:133]: ResizeAsync

File: src/ImageService.Broker/Consumers/CreateImagesConsumer.cs [hunk 3]
```diff
@@ -161,31 +171,36 @@ private object CreateMessageImages(ICreateImagesRequest request)
...
       DbImageMessage dbImageMessage;
       DbImageMessage dbPrewiewImageMessage;
-      string resizedContent;
+      (bool isSuccess, string resizedContent, string extension) resizeResult;
 
       foreach (CreateImageData createImage in request.Images)
       {
         dbImageMessage = _dbImageMessageMapper.Map(createImage);
-        resizedContent = _resizeHelper.Resize(createImage.Content, createImage.Extension);
+        resizeResult = await _resizeHelper.Resize(createImage.Content, createImage.Extension);
```
Comment (MariyaSin) [inline src/ImageService.Broker/Consumers/CreateImagesConsumer.cs:179]: ResizeAsync

File: src/ImageService.Data/ImageNewsRepository.cs [hunk 4]
```diff
@@ -38,21 +40,21 @@ public bool Remove(List<Guid> imageIds)
...
+        await _provider.ExecuteRawSqlAsync($@"DELETE FROM {DbImageNews.TableName} WHERE Id = '{imageId}' OR ParentId = '{imageId}' OR
           Id IN (SELECT ParentId FROM {DbImageNews.TableName} WHERE Id = '{imageId}' AND ParentId IS NOT NULL);");
       }
 
       return true;
     }
 
-    public List<DbImageNews> Get(List<Guid> imageIds)
+    public async Task<List<DbImageNews>> GetAsync(List<Guid> imageIds)
     {
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageNewsRepository.cs:51]: rename imagesIds an add null or empty check

File: src/ImageService.Data/ImageProjectRepository.cs [hunk 5]
```diff
@@ -38,21 +40,21 @@ public bool Remove(List<Guid> imageIds)
...
-        _provider.ExecuteRawSql($@"DELETE FROM {DbImageProject.TableName} WHERE Id = '{imageId}' OR ParentId = '{imageId}' OR
+        await _provider.ExecuteRawSqlAsync($@"DELETE FROM {DbImageProject.TableName} WHERE Id = '{imageId}' OR ParentId = '{imageId}' OR
           Id IN (SELECT ParentId FROM {DbImageProject.TableName} WHERE Id = '{imageId}' AND ParentId IS NOT NULL);");
       }
 
       return true;
     }
 
-    public List<DbImageProject> Get(List<Guid> imageIds)
+    public async Task<List<DbImageProject>> GetAsync(List<Guid> imageIds)
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageProjectRepository.cs:50]: imagesIds

File: src/ImageService.Data/ImageUserRepository.cs [hunk 6]
```diff
@@ -16,20 +18,20 @@ public ImageUserRepository(IDataProvider provider)
...
 
       _provider.ImagesUsers.AddRange(imagesUsers);
-      _provider.Save();
+      await _provider.SaveAsync();
 
       return imagesUsers.Select(x => x.Id).ToList();
     }
 
-    public bool Remove(List<Guid> imageIds)
+    public async Task<bool> RemoveAsync(List<Guid> imageIds)
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageUserRepository.cs:34]: imagesIds

File: src/ImageService.Data/Interfaces/IImageMessageRepository.cs [hunk 7]
```diff
@@ -8,12 +9,12 @@ namespace LT.DigitalOffice.ImageService.Data.Interfaces
   [AutoInject]
   public interface IImageMessageRepository
   {
-    List<Guid> Create(List<DbImageMessage> imagesMessages);
+    Task<List<Guid>> CreateAsync(List<DbImageMessage> imagesMessages);
 
-    List<DbImageMessage> Get(List<Guid> imageIds);
+    Task<List<DbImageMessage>> GetAsync(List<Guid> imageIds);
```
Comment (MariyaSin) [inline src/ImageService.Data/Interfaces/IImageMessageRepository.cs:14]: imagesIds

File: src/ImageService.Data/Interfaces/IImageUserRepository.cs [hunk 8]
```diff
@@ -8,12 +9,12 @@ namespace LT.DigitalOffice.ImageService.Data.Interfaces
   [AutoInject]
   public interface IImageUserRepository
   {
-    List<Guid> Create(List<DbImageUser> imagesUsers);
+    Task<List<Guid>> CreateAsync(List<DbImageUser> imagesUsers);
 
-    List<DbImageUser> Get(List<Guid> imageIds);
+    Task<List<DbImageUser>> GetAsync(List<Guid> imageIds);
```
Comment (MariyaSin) [inline src/ImageService.Data/Interfaces/IImageUserRepository.cs:14]: imagesIds

File: src/ImageService.Data/Interfaces/IImageUserRepository.cs [hunk 9]
```diff
@@ -8,12 +9,12 @@ namespace LT.DigitalOffice.ImageService.Data.Interfaces
...
 
-    DbImageUser Get(Guid imageId);
+    Task<DbImageUser> GetAsync(Guid imageId);
 
-    bool Remove(List<Guid> imageIds);
+    Task<bool> RemoveAsync(List<Guid> imageIds);
```
Comment (MariyaSin) [inline src/ImageService.Data/Interfaces/IImageUserRepository.cs:18]: imageIds

File: src/ImageService/Controllers/FileImageController.cs [hunk 10]
```diff
@@ -10,12 +11,12 @@ namespace LT.DigitalOffice.ImageService.Controllers
   public class FileImageController : ControllerBase
   {
     [HttpGet("get")]
-    public FileResult Get(
+    public async Task<FileResult> Get(
```
Comment (MariyaSin) [inline src/ImageService/Controllers/FileImageController.cs:14]: GetAsync

File: src/ImageService/Controllers/ProjectController.cs [hunk 11]
```diff
@@ -11,11 +12,11 @@ namespace LT.DigitalOffice.ImageService.Controllers
   public class ProjectController : ControllerBase
   {
     [HttpGet("get")]
-    public OperationResultResponse<ImageResponse> Get(
+    public async Task<OperationResultResponse<ImageResponse>> Get(
```
Comment (MariyaSin) [inline src/ImageService/Controllers/ProjectController.cs:15]: GetAsync

File: src/ImageService.Broker/Consumers/CreateImagesConsumer.cs (hunk 3 above)
Comment (ndoriomedov) [reply src/ImageService.Broker/Consumers/CreateImagesConsumer.cs:179]: will change it in lib first

**Pull request merged**